### ISSUE TITLE
Evaluate v2.0-rc2

### ISSUE THREAD

**Issue by Schluggi** [Schluggi/AIOsense#15]
We just bought some AIOsense v2.0-rc2.

They will arrive late January. 

Things we have to test:

- [x] mmWave
- [x] PIR
- [x] generic I²C ports
- [x] BME280 (onboard)
- [x] BME680 (onboard)
- [ ] onboard power
- [ ] IO pins
- CPUs
  - [x] ESP32-C3
  - [ ] ESP32-S2
  - [x] ESP32-D1
  - [ ] ESP8266-D1
- Modules
  - [x] BME280
  - [x] BME680
  - [x] BH1750

**Comment by Schluggi** [comment]
The PCBs arrived today. I'll test them within the next days.

**Comment by Schluggi** [comment]
After testing a bit around i found out that the air quality stuff of the BME680 does not work at all with an non generic ESP32. 

Moreover there is a new problem with the PIR pin. If you want to use the PIR on the v2.0-rc2 board with an ESP8266 D1 mini, you have to remove the jumper and connect the center pin to GPIO13/MOSI. That's because GPIO0/D3 and GPIO2/D4 are always up and used for selecting the boot-mode.

**Comment by s-gordon** [comment]
Thanks for the update @Schluggi! I was a little impatient and had some of the v2.0-rc2 boards fabricated as well. Can confirm that routing the PIR data pin to GPIO13 works for me as well.

What's the thinking here? Modify the design so that the jumper pin routes to GPIO13?

**Comment by Schluggi** [comment]
> What's the thinking here? Modify the design so that the jumper pin routes to GPIO13?

We have notices some more problems with our current design:

- `bluetooth_proxy` only works with the arduino framework which is not compatible with non generic ESP32 boards ([source](https://esphome.github.io/bluetooth-proxies/))
- `web_server` and `captive_portal` only works with the arduino framework as well (would be nice to have)
- `bme680_bsec` are also not compatible. Without these module you cannot get values for IAQ, CO2 Equivalent & VOC Equivalent (temperature, humidity and pressure will work via the `bme680` module) 

I would drop the ESP32-C3/S2 and ESP8266 compatibility and go with the generic ESP32. But this would have major impact on the project. @MeisterGig is against it. 

We're happy to hear your opinion to this topic.

**Comment by Schluggi** [comment]
After some more research from @MeisterGig he found out that `bluetooth_proxy` is actually working on the ESP32-C3 (since a big change in [ESPHome 2022.12](https://esphome.io/changelog/2022.12.0.html#bluetooth-proxy)).

He also found a [pull-request](https://github.com/esphome/esphome/pull/3500) for the `web_server` and `captive_portal` problem.

So `bme680_bsec` is the only module that's still not working. Maybe we can find a solution here as well.

**Comment by florianmulatz** [comment]
> 680 does not work at all with an non generic ESP32.
> 
> Moreover there is a new problem with the PIR pin. If you want to use the PIR on the v2.0-rc2 board with an ESP8266 D1 mini, you have to remove the jumper and connect the center pin to GPIO13/MOSI. That's because GPIO0/D3 and GPIO2/D4 are always up and used for selecting the boot-mode.

Do I understand correctly that a generic ESP32 (aka WEMOS D1 form-factor like this --> [ESP32 - AMZ Link](https://www.amazon.de/dp/B08BTYCGVV?ref=ppx_yo2ov_dt_b_product_details&th=1) will work without any problems including PIR and I2C Dev-Boards (like f.e. this one [URL](https://www.joy-it.net/de/products/SEN-BME680) )

Cheers

**Comment by Schluggi** [comment]
> Do I understand correctly that a generic ESP32 (aka WEMOS D1 form-factor like this --> [ESP32 - AMZ Link](https://www.amazon.de/dp/B08BTYCGVV?ref=ppx_yo2ov_dt_b_product_details&th=1) will work without any problems including PIR and I2C Dev-Boards (like f.e. this one [URL](https://www.joy-it.net/de/products/SEN-BME680) )
> 
> If so - I highly appreciate to go for ESP32 (as mentioned above). It's anyhow the way to go for most projects (as of today :) )

On a software-side the generic ESP32 (aka WEMOS D1) works the best because we can use the slow, old but reliable arduino-framework inside ESPHome. There is a newer and faster framework called "esp-idf". But unfortunately not all modules are supported right now. 

The WEMOS D1 generic ESP32 Board has a different pin layout then the ESP8266, ESP32-C3 and ESP-S2, so it will not work with any version of AIOsense yet. 

The I²C dev-boards will work with every version of AIOsense but depending of the ESP the feature set of the BME680 is different. If you want to get all out of your BME680 use an ESP8266 and the v2.0-rc3 release. But because its a ESP8266, it has no bluetooth. The PIR will work with [this](https://github.com/Schluggi/AIOsense/issues/15#issuecomment-1382575615) modification on every board v2.0-rc2 and above. 

The ESP32-C3 has some benefits over the generic ESP32 (USB Type-C, no programmer needed). We plan also to release a SMD only version of AIOsense. So we should have this in mind. No need for a programmer chip is very nice and cuts down the costs.

So i opened an [ESPHome feature request](https://github.com/esphome/feature-requests/issues/2072) to support the ESP32-C3. This would be the best solution in my opinion. If there is no progress until the end of January, we will switch completely to the generic ESP32.

**Comment by florianmulatz** [comment]
> The WEMOS D1 generic ESP32 Board has a different pin layout then the ESP8266, ....

I disagree :) From my point of perspective they're pin-compatible (I can not say anything about ESP-C3 / ESP-S2 - since I don't konw (neither own) them)

![image](https://user-images.githubusercontent.com/33655308/213625758-90c256b1-f468-4743-a11e-c6d49a619df6.png)

Please just let me know if I get this wrong :)

**Comment by s-gordon** [comment]
> > The WEMOS D1 generic ESP32 Board has a different pin layout then the ESP8266, ....
> 
> I disagree :) From my point of perspective they're pin-compatible (I can not say anything about ESP-C3 / ESP-S2 - since I don't konw (neither own) them)

Agree with @florianmulatz. I've tried this with a generic ESP32 (MH-ET Live MiniKit) and it seems fine. I've only tested THT BME280, BH1750, SEN0395, and the PIR module. Happy to lodge a PR with my YAML if there's interest.

**Comment by Schluggi** [comment]
> > > The WEMOS D1 generic ESP32 Board has a different pin layout then the ESP8266, ....
> > 
> > 
> > I disagree :) From my point of perspective they're pin-compatible (I can not say anything about ESP-C3 / ESP-S2 - since I don't konw (neither own) them)
> 
> Agree with @florianmulatz. I've tried this with a generic ESP32 (MH-ET Live MiniKit) and it seems fine. I've only tested THT BME280, BH1750, SEN0395, and the PIR module. Happy to lodge a PR with my YAML if there's interest.

I totally agree. I wonder how i could miss that. Can you upload an image of the ESP32 on the board? PRs are always welcome :)

I just ordered a few ESP32 mini boards to test it myself. 


Moreover I changed the labels of the ports. Now you can see more easily whats connected to. That should help by finding the right port in the ESPHome-config. Any comments on that?

![image](https://user-images.githubusercontent.com/43608073/213885606-bf02e85e-ec28-4a5b-b23d-15af3a0973d8.png)


Next steps:
I will test the ESP32 D1 mini board. If everything works fine I will finally release v2.0. Thank you for your help :)

**Comment by s-gordon** [comment]
Sure. These aren't the best pictures, but here you go :-). I salvaged the ESP32 board from another project and needed to remove the old header pins - hence why the board looks a little worn. I'm waiting on the PIR sensors to arrive later this week, and so they aren't shown here.

No worries. I'll look into doing a PR later in the week.

![esp32](https://user-images.githubusercontent.com/4898021/213905883-d829c60f-fa11-4698-8932-b312a4dc0967.jpg)
![board](https://user-images.githubusercontent.com/4898021/213905887-3ed78b70-0706-45cb-9418-44384e02abd0.jpg)

**Comment by Schluggi** [comment]
@s-gordon 

Thanks! :)
Do I see it right that the ESP32 mini is bigger than the AIOsense board?

**Comment by s-gordon** [comment]
That's right, it does protrude beyond the bottom edge of the AIOsense board by about 3 mm. The camera perspective exaggerates it a little.

**Comment by Schluggi** [comment]
I finally found a way to use the `bme680_bsec` component with an ESP32-C3 (fb7976eb35cdcf3bc050d74ac4c29ba4ce0aaad7). 🥳 

Everything works now!

So the only thing we have to discuss: Does we want to increase the board size to fit better with generic ESP32 boards? I would say no, because otherwise we would have a problem with the case.

But maybe @MeisterGig could come up with a case design option for generic ESP32 boards? So we don't have to increase the board size, just the case?

**Comment by s-gordon** [comment]
I'm in favour of increasing the PCB size to accommodate generic ESP32 boards, although I am being a little selfish :-P.

If reworking the case design is a concern, then I'd be happy to contribute the STLs that I've put together. This will fit both the D1 Mini and generic ESP32.

**Comment by MeisterGig** [comment]
I agree with @Schluggi. It is unwise to increase the size of the board even more. It's already quite big. As far as i can see the length would also need to be modified, this would be a disadvantage for c3 and s2 modules, because the USB port would be too far in the case.

In addition, c3 and s2 chips are newer, cheaper and will be supported for a longer time compared to the 32 series https://www.espressif.com/en/products/longevity-commitment

**Comment by MeisterGig** [comment]
I agree with @Schluggi. It is unwise to increase the size of the board even more. It's already quite big. As far as i can see the length would also need to be modified, this would be a disadvantage for c3 and s2 modules, because the USB port would be too far in the case.

In addition, c3 and s2 chips are newer, cheaper and will be supported for a longer time compared to the 32 series https://www.espressif.com/en/products/longevity-commitment

If we are looking a little bit further in the future. Espressif just launched the ESP32-C6 (which has WiFi 6 and Zigbee support) https://www.espressif.com/en/products/socs/esp32-c6 this chip also has a similar size as the c3 and comes with an internal USB to Serial module like the c3 and S2. So my best guess is, that WEMOS will probably design the c6-mini in the same form factor as the c3 and s2 mini.

**Comment by florianmulatz** [comment]
Agree to. Although I've more esp32 (d1mini) at home - I'm willing to invest in a more modern platform as well of course. ESP32-c6 sounds cool - ZIGBEE let's my heart beats faster 😅

**Comment by Schluggi** [comment]
> Although I've more esp32 (d1mini) at home - I'm willing to invest in a more modern platform as well of course. 

You don't have to. The generic ESP32 will be compatible. We just don't increase the board size for this one. So it overlaps as @s-gordon pictures has shown. @MeisterGig will also release a case design for this one. So you can still use your generic ones. However the C3 will fit better.

Yes, the C6 looks cool but please have in mind we have to wait until ESPHome and Bosch supports it. 


At this moment I will release v2.0-rc4 in the next days and order these. Afterwards we hopefully have a bug free v2 release :)
